compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -123,6 +136,7 @@ class ScalaSettings extends Settings.SettingGroup with CommonScalaSettings {
   val XignoreScala2Macros: Setting[Boolean] = BooleanSetting("-Xignore-scala2-macros", "Ignore errors when compiling code that calls Scala2 macros, these will fail at runtime.")
   val XimportSuggestionTimeout: Setting[Int] = IntSetting("-Ximport-suggestion-timeout", "Timeout (in ms) for searching for import suggestions when errors are reported.", 8000)
   val Xsemanticdb: Setting[Boolean] = BooleanSetting("-Xsemanticdb", "Store information in SemanticDB.").withAbbreviation("-Ysemanticdb")
+  val Xtarget: Setting[String] = ChoiceSetting("-Xtarget", "target", "Emit bytecode for the specified version of the Java platform. This setting is ignored if -release is specified.", supportedTargetVersions, supportedTargetVersions.head) withAbbreviation "--Xtarget"
```

Comment:
> This setting is ignored if -release is specified.

That's a bit confusing, I think it's fine to let `-Xtarget` override the target set with `-release`.

Reply:
For javac you get an error if you try to use `-source` and `-target` at the same time. That would be the best option IMHO but I'm not sure if this can be easily achieved with our current approach to parsing command line options.
What would be the purpose of using `-source` and `-Xtarget` at the same time with different values? Then running scalac on JDK X you would try to compile code with classes from JDK Y on classpath to emit bytecode to be run on JDK Z, which would be quite likely to break at runtime. The only use case for using `-Xtarget` alone seems to be cross-compiling at your own risk for a newer JDK than the one used for the compilation.

Reply:
> For javac you get an error if you try to use -source and -target at the same time. That would be the best option IMHO but I'm not sure if this can be easily achieved with our current approach to parsing command line options.

That or an error would be fine too, that could be checked in `lazy val classfileVersion`.

Reply:
javac also emits a warning when you use `-target` explicitly but there is a mismatch with the current JDK version. Should we have it as well? (I guess we don't want to raise a warning for the default when compiling on JDK 9+)

Reply:
Since we renamed it to `-Xtarget` it's an advanced option so I don't think a warning is worth it, but the documentation of Xtarget can mention that people should consider using release instead